google/cloud/bigtable/testing/table_integration_test.cc

```diff
@@ -30,6 +31,57 @@ std::string TableTestEnvironment::instance_id_;
 std::string TableTestEnvironment::cluster_id_;
 std::string TableTestEnvironment::zone_;
 std::string TableTestEnvironment::replication_zone_;
+google::cloud::internal::DefaultPRNG TableTestEnvironment::generator_;
+std::string TableTestEnvironment::table_id_;
+
+void TableTestEnvironment::SetUp() {
+  generator_ = google::cloud::internal::MakeDefaultPRNG();
+
+  auto admin_client = bigtable::CreateDefaultAdminClient(
+      TableTestEnvironment::project_id(), ClientOptions());
+  auto table_admin =
+      bigtable::TableAdmin(admin_client, TableTestEnvironment::instance_id());
+
+  std::string const family1 = "family1";
+  std::string const family2 = "family2";
+  std::string const family3 = "family3";
+  std::string const family4 = "family4";
+  bigtable::TableConfig table_config =
+      bigtable::TableConfig({{family1, bigtable::GcRule::MaxNumVersions(10)},
+                             {family2, bigtable::GcRule::MaxNumVersions(10)},
+                             {family3, bigtable::GcRule::MaxNumVersions(10)},
+                             {family4, bigtable::GcRule::MaxNumVersions(10)}},
+                            {});
+
+  table_id_ = RandomTableId();
+  ASSERT_STATUS_OK(table_admin.CreateTable(table_id_, table_config));
+}
+
+void TableTestEnvironment::TearDown() {
+  auto admin_client = bigtable::CreateDefaultAdminClient(
+      TableTestEnvironment::project_id(), ClientOptions());
+  auto table_admin =
+      bigtable::TableAdmin(admin_client, TableTestEnvironment::instance_id());
+
+  ASSERT_STATUS_OK(table_admin.DeleteTable(table_id_));
+}
+
+std::string TableTestEnvironment::CreateRandomId(std::string const& prefix,
+                                                 int count) {
+  return prefix +
+         google::cloud::internal::Sample(
+             generator_, count, "abcdefghijklmnopqrstuvwxyz0123456789");
+}
+
+std::string TableTestEnvironment::RandomTableId() {
+  constexpr int RANDOM_CHARACTERS = 16;
+  return CreateRandomId("table-", RANDOM_CHARACTERS);
+}
+
+std::string TableTestEnvironment::RandomInstanceId() {
+  constexpr int RANDOM_CHARACTERS = 8;
+  return CreateRandomId("it-", RANDOM_CHARACTERS);
```

Comment:
"it", being short for "instance"?

Reply:
Yup.  I moved this code from elsewhere btw, as I recall the maximum is about 12 , and we want as many as possible for the random portion.

Reply:
So I was wrong, the maximum length is 33 characters (the only place this seems to be documented in is the Web UI, :shrug:). With that much room it makes no sense to abbreviate.